docs/curriculum/Versions.md

```diff
@@ -1,10 +1,305 @@
+using the deprecated TaskQueue. If a single WorkerFactory is in a process, remove it from
+the deployment.
+
+## Replay Tests
+
+Recall that Workflow Executions are "replayed" at various times during your Application lifetime.
+
+> This means we must verify that the state and behaviors meet expectations regardless of **how** it is executed.
+
+Your Workflow could build, pass all unit and functional tests, and ship to production just fine, 
+but still bring your business to a halt because of Non-Determinism Errors.
+This incident will be now exacerbated by the _new_ Workflow Executions that have started in production while the previous versions were "stuck".
+
+The Temporal [WorkflowReplayer](https://docs.temporal.io/develop/java/testing-suite#replay) test facility is what you want to use to either verify or validate any production Workflow
+executions in your development or build pipeline.
+
+These types of tests are a variant of "smoke" or "build validation" tests, so run these tests where you are currently performing these other tests; 
+likely this is a step just before running your other unit or functional tests.
+Alternately, you can run these tests as a *verification* step and might put these in the developer code commit workflow as early as a `git commit`.
+
+Regardless of your environment or where in the delivery process these appear, they are an important investment to your production quality.
+
+#### Recommendations
+
+Replay tests should exercise history that was produced by various code paths through your Executions.
+If your Workflow Definition has conditional branches, loops, or timers, it makes sense to store histories created by the unit test cases that verify these scenarios with the
+execution histories _caused_ by the scenarios in the `latest` build so that they might be validated in `vNext` implementations.
+
+Factors which determine how these tests are executed are based on your Workflow Version strategy and isolation boundaries present in your
+Continuous Integration / Continuous Deployment (CICD) pipeline.
+
+
+#### Implementing Replay Tests : Patched Version Strategy
+
+If you elect to use the `GetVersion` or `patch` SDK APIs, you must choose between loading Workflow History representing the
+`latest` build from memory or from file storage.
+
+**External Workflow History Storage**
+
+* _Cons_
+    * This can be problematic in ephemeral containers used for CICD purposes.
+    * This might require non-trivial authnz to an external file storage solution.
+    * Garbage collection on histories might be needed to keep storage costs down
+        * Detecting which histories are eligible for collection can be complex for Workflow Definitions that are long-running and do not implement Continue As New
+    * The `DataConverter` used to generate the histories _must_ be used when doing these tests. 
+* _Pros_
+    * No need to maintain more than one implementation in VCS for the same Workflow Type in the same source version.
```

Comment:
```suggestion
    * No need to maintain more than one implementation in a version control system (VCS) for the same Workflow Type in the same source version.
```

I suspect that some developers might not be familiar with the abbreviation, so I recommend spelling it out on first reference here. In my experience, some (older) people say "revision control system." I suspect that some younger people have only ever known git and therefore have no need to generalize about what kind of tool it is.